symengine/dict.h

```diff
@@ -185,6 +185,165 @@ std::ostream &operator<<(std::ostream &out, const SymEngine::vec_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::set_basic &d);
 std::ostream &operator<<(std::ostream &out, const SymEngine::map_int_Expr &d);
 
+// dict wrapper
+template <typename Key, typename Value, typename Wrapper>
+class ODictWrapper
```

Comment:
It only acts like a polynomial because that's what the operators like `*` and `+` do. So, I think it should be in the polynomial file.